### PR TITLE
WCMS-1206: advertisement middleware

### PR DESCRIPTION
**Pull request by klipach**
https://nzzdev.atlassian.net/browse/WCMS-1206

### CODE REVIEW

File: src/middleware/pre-advertisement-adder.coffee [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports =
+
+  handler: (livingdoc) ->
+    advertisementComponent = livingdoc.createComponent('advertisement')
+    barrierComponent = livingdoc.componentTree.find('barrier')[0]
+
+    if barrierComponent
+      barrierComponent.after(advertisementComponent)
```
Comment (DaRaFF) [inline src/middleware/pre-advertisement-adder.coffee]: * Are you sure that having this advertisement component after the barrier component is correct? Have you talked with @yoyurec about that?
* Is it correct not to add ads, when no barrier component is there? ads should always be added and should have no dependency to something else.

**Comment by DaRaFF** [comment]
review done

**Comment by klipach** [comment]
fixed

File: test/middleware/pre-advertisement-adder.coffee [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  t.plan(1)
+  t.equal('pre', middleware.scope, 'has "pre" scope')
+
+
+test 'Advertisement adder has correct handler', (t) ->
+  t.plan(1)
+  t.assert(typeof middleware.handler == 'function', 'handler is a function')
+
+
+test 'Advertisement adder adds advertisement marker after barrier', (t) ->
```
Comment (DaRaFF) [inline test/middleware/pre-advertisement-adder.coffee]: before

**Pull request merged**